staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go

```diff
@@ -894,7 +901,12 @@ func (vs *VSphere) LoadBalancer() (cloudprovider.LoadBalancer, bool) {
 }
 
 func (vs *VSphere) isZoneEnabled() bool {
-	return vs.cfg != nil && vs.cfg.Labels.Zone != "" && vs.cfg.Labels.Region != ""
+	isEnabled := vs.cfg != nil && vs.cfg.Labels.Zone != "" && vs.cfg.Labels.Region != ""
+	if isEnabled && vs.isSecretInfoProvided && vs.nodeManager.credentialManager == nil {
```

Comment:
Why is this exactly like that? Should not informers and credentialManager eventually set up and stop being nil? Because returning false here is permanent - no other consumer will be able to initialize node with this labels

Reply:
Please add a comment specifying that such combination of values when `vs.isSecretInfoProvided && vs.nodeManager.credentialManager != nil` is unique to KCM only. So that it will be clear this code will return false only in kubelet.

Reply:
Kubelet will never have this `credentialManager`, informers doesn't setting up there. This thing just for workaround it and prevent kubelet from hanging up during startup if credentials stored in secret.

Within KCM or with plain text credentials all works as before.